File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/view/parameters/PageScreenData.java
package com.esofthead.mycollab.module.project.view.parameters;

import com.esofthead.mycollab.module.wiki.domain.Page;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

/**
 * 
 * @author MyCollab Ltd.
 * @since 4.4.0
 *
 */
public class PageScreenData {

	public static class Read extends ScreenData<Page> {

		public Read(Page params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<Page> {

		public Add(Page component) {
			super(component);
		}
	}

	public static class Edit extends ScreenData<Page> {

		public Edit(Page component) {
			super(component);
		}
	}

	public static class Search extends ScreenData<String> {

		public Search(String baseFolder) {
			super(baseFolder);
		}
	}
}

File: mycollab-web/src/main/java/com/mycollab/module/project/view/parameters/PageScreenData.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.module.project.view.parameters

import com.mycollab.module.page.domain.Page
import com.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object PageScreenData {
    class Read(params: Page) : ScreenData<Page>(params)

    class Add(params: Page) : ScreenData<Page>(params)

    class Edit(params: Page) : ScreenData<Page>(params)

    class Search(params: Any?) : ScreenData<Any>(params)
}